rust/eos-rs/src/api/dungeon_mode/effects.rs

```diff
@@ -1088,21 +1013,46 @@ impl DungeonEffectsEmitter {
     /// the Mold Breaker ability on the user.
     pub fn try_switch_place(&self, attacker: &mut DungeonEntity, defender: &mut DungeonEntity) {
         // SAFETY: We have a lease on the overlay existing.
-        unsafe { ffi::TrySwitchPlace(attacker as *mut _, defender as *mut _) }
+        unsafe { ffi::TrySwitchPlace(attacker, defender) }
+    }
+
+    /// Cures the target's freeze, shadow hold, ingrain, petrified, constriction or wrap (both as
+    /// user and as target) status due to the action of the user.
+    ///
+    /// # Arguments
+    /// * `attacker` - The monster that is trying to inflict this status.
```

Comment:
Here the attacker is not inflicting the status, it's performing the action that's causing the status to be forcefully cured (for example, the monster using a fire-type move on a frozen target)